lib/puppet/provider/swap_file/linux.rb

```diff
@@ -0,0 +1,110 @@
+Puppet::Type.type(:swap_file).provide(:linux) do
+
+  confine  :kernel   => :linux
+  commands :swapon   => 'swapon'
+  commands :swapoff  => 'swapoff'
+  commands :mkswap   => 'mkswap'
+
+  mk_resource_methods
+
+  def initialize(value={})
+    super(value)
+    @property_flush = {}
+  end
+
+  def self.get_swap_files
+    swapfiles = swapon(['-s']).split("\n")
+    swapfiles.shift
+    swapfiles.sort
+  end
+
+  def self.prefetch(resources)
+    instances.each do |prov|
+      if resource = resources[prov.name]
+        resource.provider = prov
+      end
+    end
+  end
+
+  def self.instances
+    get_swap_files.collect do |swapfile_line|
+      swapfile_properties = get_swapfile_properties(swapfile_line)
+      new(swapfile_properties)
+    end
+  end
+
+  def self.get_swapfile_properties(swapfile_line)
+    swapfile_properties = {}
+
+    # swapon -s output formats thus:
+    # Filename        Type    Size  Used  Priority
+
+    # Split on spaces
+    output_array = swapfile_line.strip.split(/\s+/)
+
+    # Assign properties based on headers
+    swapfile_properties[:ensure]     = :present
```

Comment:
I'd just write this out as a hash and save on the typing of swapfile_properties:

```
swapfile_properties = {
  name: output_array[0].
  ...
}
```

Bonus bonus if you just have an array of keys given the progression on the value side.